Feature/rafael sapalo/#88/setup postgres container

Rafael-Sapalo opened the pull request.

Rafael-Sapalo committed:
feat: add PostgreSQL database integration to auth and chat services

- Added PostgreSQL service with health checks and database initialization
- Created init_dbs.sql script to initialize elyrii_users, elyrii_journal, and elyrii_chats databases
- Added database environment variables (DB_HOST, DB_USER, DB_PASS, DB_NAME, DB_PORT) to auth and chat services
- Updated auth service Dockerfile to run database migrations on startup
- Changed default database name from 'elyrii' to 'elyrii_users' in auth service configuration

Rafael-Sapalo committed:
feat: improve auth service configuration and database setup

- Changed Docker network name from 'elyrii' to 'elyrii_net' across all services for better naming convention
- Added SECRET_KEY environment variable to auth service configuration
- Updated auth service Dockerfile to run db:generate, db:migrate, and start in single ENTRYPOINT command
- Removed SSL requirement from PostgreSQL connection string for local development
- Changed user table name from 'user' to 'users' for SQL naming conventions

Mord0rak reviewed (changes requested):
creation d'un fichier d'environement pour la prod et le dev stp

Mord0rak commented on elyrii_server/services/auth/src/config/db/user.table.ts:
```diff
@@ -3,7 +3,7 @@ import { integer, pgTable, text, timestamp, uuid } from "drizzle-orm/pg-core";
  * Drizzle ORM table definition for users authenticated by the auth service.
  */
-export const userTable = pgTable("user", {
+export const userTable = pgTable("users", {
```
.index() pour l'email afin de garantir l'indexation

Rafael-Sapalo committed:
feat: add environment variable configuration and database indexing to backend services

- Added .gitignore file for elyrii_server to exclude environment files, node_modules, build artifacts, logs, editor files, Docker files, and documentation
- Replaced hardcoded values with environment variables across all services (gateway, auth, chat, postgres)
- Added GATEWAY_PORT and GATEWAY_SECRET_KEY environment variables to gateway service
- Updated auth service to use AUTH_PORT, DB_HOST, DB_USER, DB_PASSWORD

Rafael-Sapalo committed:
feat: add environment variable configuration and database indexing to backend services

- Added environment variables to backend-build workflow for database, gateway, and JWT configuration
- Changed database name environment variables from generic DB_NAME to service-specific AUTH_DB_NAME and CHAT_DB_NAME
- Updated docker-compose.yml to use AUTH_DB_NAME for auth service and CHAT_DB_NAME for chat service
- Removed concurrent index creation from user table email index for compatibility
- Added buil

Rafael-Sapalo committed:
feat: add environment variable validation for service URLs in gateway

- Removed fallback values for CHAT_SERVICE_URL and AUTH_SERVICE_URL environment variables
- Added validation checks to throw errors if CHAT_SERVICE_URL or AUTH_SERVICE_URL are not defined
- Ensures gateway service fails fast on startup if required service URLs are missing
